jumpscale/tools/redis/redis.py

```diff
@@ -25,8 +25,10 @@ def start(self, host: str = "127.0.0.1", port: int = 6379):
             host (str, optional): redis bind address. Defaults to "127.0.0.1".
             port (int, optional): redis port. Defaults to 6379.
         """
-        self.cmd.start_cmd = f"redis-server --bind {host} --port {port}"
-        self.cmd.start()
+        # Port is not busy (Redis is not started)
+        if not j.sals.nettools.tcp_connection_test("127.0.0.1", 6379, timeout=1):
+            self.cmd.start_cmd = f"redis-server --bind {host} --port {port}"
+            self.cmd.start()
```

Comment:
should add port/process string/reges to be able to hold the process for later stop

Reply:
check nginx or threebot entry point